manual/parts/threads.tex

```diff
@@ -72,6 +74,21 @@ \subsection{Thread Deactivation}
 deleted.
 % an example of which is demonstrated in \nameref{ex:second_thread}.
 
+\subsection{Core Affinity}
+\label{sec:thread_affinity}
+
+On the standard (non-MCS) kernel in a SMP configuration, the core affinity of
+a thread can be changed anytime using \apifunc{seL4\_TCB\_SetAffinity}{tcb_setaffinity}.
+A thread can change it own affinity, if it has access to it's \obj{TCB} object.
+Affinity changes triggers a rescheduling of the target thread. If the new
+affinity is the same than the old affinity, the behaviour is effectively similar
+to what would happen if the target thread did a \apifunc{seL4\_Yield}{sel4_yield}.
```

Comment:
```suggestion
affinity is the same as the old affinity, the behaviour is effectively similar
to what would happen if the target thread did a \apifunc{seL4\_Yield}{sel4_yield}.
```

"effectively similar" is not very useful, because it doesn't give people any certainty on what actually happens. We. should either commit to that it really is effectively a  `yield` operation, or we should leave out that part and leave unspecified what happens.

Reply:
Thanks for the feedback. I made it this blurry because I was not sure if there is a defined behaviour at all. That is just what the implementation roughly does. Is there a formal spec here? Otherwise, what is a wording to state what to expect in the worst case. Saying "undefined" is also not helpful to set some informal limits. Can it be worse than 'yield'?

Reply:
For SMP, the manual is the only spec, that's why it matter a bit more what we say here -- I would in this case probably just not talk about this case, because the operation will do what is said before (set the affinity), and whether this comes out to a yield or not in total if the affinity hasn't changed is not really important. If we commit here to it being the same we'd have to make sure that the implementation covers the same corner cases etc and people might start relying on it.

Reply:
Exactly what @lsf37 said: You don't want people to count on behavioural details like these. There is no practical use for these kind of details, please don't document them, that only lowers the signal to noise ration in the manual.

Please drop the whole new section, it's mostly redundant: non-MCS is clear as it is, especially after your improvement. If MCS needs further explanation it should be done in the Scheduling Context Creation chapter and maybe add a reference.

Reply:
I think it worth keeping the note that affinity values are platform specific and there is a mapping. Once does not pass the actual physical ID here.

Reply:
Affinity values are a platform specific mapping by definition, between some random number and a physical core. Pointing out the obvious doesn't help any one, so I'm still in favour of dropping the section as it is.

If you mean that on some RISCV platforms the core IDs are not `0...N` then be specific and just state that instead (and personally I would consider that a bug, but oh well). If you mean that the startup task's affinity is platform specific, then that could be mentioned somewhere too.